version: '3.8'

services:
  redis:
    image: redis:latest
    ports:
      - "6379:6379"
    networks:
      - bot-network

  postgres:
    image: postgres:latest
    environment:
      POSTGRES_DB: postgres
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: qwaszx12
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    networks:
      - bot-network

  django:
    build: .
    command: >
      sh -c "python manage.py migrate &&
             python manage.py createsuperuser --noinput --username admin --email admin@example.com &&
             python manage.py runserver 0.0.0.0:8000"
    ports:
      - "8000:8000"
    volumes:
      - .:/app
    env_file:
      - .env
    depends_on:
      - redis
      - postgres
    networks:
      - bot-network

  celery:
    build: .
    command: celery -A app worker --loglevel=info --pool=solo
    volumes:
      - .:/app
    environment:
      - REDIS_URL=redis://redis:6379/0
      - DATABASE_URL=postgres://postgres:qwaszx12@postgres:5432/postgres
    depends_on:
      - redis
      - postgres
    networks:
      - bot-network

  bot:
    build: .
    command: python -m app.bot.main
    volumes:
      - ./Orders.xlsx:/app/Orders.xlsx
    environment:
      - REDIS_URL=redis://redis:6379/0
    depends_on:
      - redis
    networks:
      - bot-network

networks:
  bot-network:
    driver: bridge

volumes:
  postgres_data: